src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -29,294 +29,17 @@ internal class SilentRequest : RequestBase
             : base(serviceBundle, authenticationRequestParameters, silentParameters)
         {
             _silentParameters = silentParameters;
-        }
-
-        private bool BrokerIsInstalledAndSupportsSilentAuth
-        {
-            get
-            {
-                if (_canExecuteBrokerValue == null)
-                {
-                    _canExecuteBrokerValue = AuthenticationRequestParameters.IsBrokerConfigured && ServiceBundle.PlatformProxy.CanBrokerSupportSilentAuth();
-                }
-
-                return (bool)_canExecuteBrokerValue;
-            }
-        }
-
-        private async Task<IAccount> GetSingleAccountForLoginHintAsync(string loginHint)
-        {
-            var accounts = await CacheManager.GetAccountsAsync(ServiceBundle.Config.AuthorityInfo.CanonicalAuthority)
-                .ConfigureAwait(false);
-
-            accounts = accounts
-                .Where(a => !string.IsNullOrWhiteSpace(a.Username) &&
-                       a.Username.Equals(loginHint, StringComparison.OrdinalIgnoreCase))
-                .ToList();
-
-            if (!accounts.Any())
-            {
-                throw new MsalUiRequiredException(
-                    MsalError.NoAccountForLoginHint,
-                    MsalErrorMessage.NoAccountForLoginHint,
-                    null,
-                    UiRequiredExceptionClassification.AcquireTokenSilentFailed);
-            }
-
-            if (accounts.Count() > 1)
-            {
-                throw new MsalUiRequiredException(
-                    MsalError.MultipleAccountsForLoginHint,
-                    MsalErrorMessage.MultipleAccountsForLoginHint,
-                    null,
-                    UiRequiredExceptionClassification.AcquireTokenSilentFailed);
-
-            }
-
-            return accounts.First();
-        }
-
-        private async Task<IAccount> GetAccountFromParamsOrLoginHintAsync(AcquireTokenSilentParameters silentParameters)
-        {
-            if (silentParameters.Account != null)
-            {
-                return silentParameters.Account;
-            }
-
-            return await GetSingleAccountForLoginHintAsync(silentParameters.LoginHint).ConfigureAwait(false);
+            strategy = SilentAuthStrategyFactory.GetSilentAuthStrategy(this, serviceBundle, authenticationRequestParameters, silentParameters);
```

Comment:
According to the spec, MSAL should: 

- try the client (cache) strategy first
- if that fails, try the broker strategy second (if the broker is configured and enabled etc.)